### DISCUSSION TITLE
Is the provided Home Manager module going to be deprecated (since Hyprland is already supported by upstream Home Manager? (drift detected!)

### DISCUSSION THREAD

**Discussion by panchoh** [hyprwm/Hyprland#3252] [Q&A] [answered]
Hi, fellow Hyprlanders.

I use Hyperland via flake, to be able to participate from its development.

Until minutes ago, I configured it via the Home Manager module provided by the Hyprland flake.

A few hours ago, stylix [added support](https://github.com/danth/stylix/issues/144) for Hyperland (yay!), but this broke my flake.

After digging a little bit, I found out that (upstream) Home Manager seems to be evolving independenly of Hyprland HM module. For instance, they provide [settings](https://github.com/nix-community/home-manager/blob/f7848d3e5f15ed02e3f286029697e41ee31662d7/modules/services/window-managers/hyprland.nix#L77), which allows to express the Hyprland config as a nix expression, as an alternative to providing `extraConfig` lines to the config file.

The stylix implementation makes use of this `settings`, which is not to be found in the HM module provided by Hyperland.
There is at least another difference, namely the deprecation of `recommendedEnvironment` toggle, which is now set to always on.

So, I've been able to make my flake work again by not using the Hyprland provided HM module and relying on upstream HM, but it left me wondering (and hence this question) about the path forward:

1. is the provided HM module being deprecated in favor of HM upstream support for Hyprland and I missed the announcement?

or

2. is this a case of unplanned drift, and such discrepancies should be reported and/or fixed?

Thanks for your time, and happy hacking!

**Comment by fufexan** [comment, marked as answer]
Hi!

The upstream HM module was made by me, but in hindsight it's not as complete as we'd like. That's why @spikespaz has been hard at work to create a new module that will replace both the one currently in this flake, and the upstream HM one. The discussion was started here https://github.com/hyprwm/hyprland-wiki/pull/273#issuecomment-1647417182.

Until further updates, users are free to use whichever module suits them better.

I will make an announcement when we're planning to change the module.

**Comment by spikespaz** [comment]
If you'd like, you're free to use https://github.com/spikespaz/hyprland-nix.

The readme there should get you started. It isn't complete, and has a couple problems, but I think the `config` option is final and stable. The higher level abstractions over it are not. Innards will be rewritten however, and those changes shouldn't make a difference to you.

I should be freed up in approximately 5 days to work on it more. It will be moved to the `hyprwm` organization when ready for adoption.

**Comment by panchoh** [comment]
I think this is OBE at this point (lua, I'm looking at you ;-), so, closing.
